src/library/scala/collection/mutable/ListBuffer.scala

```diff
@@ -61,6 +62,7 @@ class ListBuffer[A]
   // Avoids copying where possible.
   override def toList: List[A] = {
     aliased = nonEmpty
+    ScalaRunTime.releaseFence()
     first
```

Comment:
Echoing @adriaanm's comments from #7028 that belong over in this PR.

> Why no fence for the other methods that mutate? I see they're private, but they are called from public methods, I assume?

My rationale is that this is the first place the list escapes. My assumption is that users are only using the `ListBuilder` itself on a single thread, and its only the resulting `List` that needs to be safely publishible.

Reply:
I'm happy to add the fence after each mutation of `tail`, at least initially, to make things easier to reason about.